dspace-server-webapp/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -441,7 +442,15 @@ public ResourceSupport findRel(HttpServletRequest request, HttpServletResponse r
         throws HttpRequestMethodNotSupportedException {
         checkModelPluralForm(apiCategory, model);
         DSpaceRestRepository<RestAddressableModel, ID> repository = utils.getResourceRepository(apiCategory, model);
-        RestAddressableModel modelObject = repository.createAndReturn();
+
+        String parentCommunityString = request.getParameter("parent");
```

Comment:
What about just having `parentUuid` be an arg to this method? Then caller (post(..)) could pass it in, having gotten it from a `@RequestParam` argument (with optional = true). It would make the contract here "You can optionally pass in a parent dso uuid") more explicit without having to read the body of the code.